src/filezcache.erl

```diff
@@ -294,14 +303,20 @@ lookup_file(Key, Opts) ->
 
 -spec delete(Key) -> Result when
 	Key :: term(),
-	Result :: ok | {error, lockedlog_a}.
+	Result :: ok | {error, Reason},
+    Reason :: lockedlog_a | writing | active.
 delete(Key) ->
-    {ok,_} = filezcache_entry_manager:delete(Key),
-    case filezcache_store:lookup(Key) of
+    Result = case filezcache_store:lookup(Key) of
         {ok, Pid} ->
             filezcache_entry:delete(Pid);
         {error, enoent} ->
             ok
+    end,
+    case Result of
+        ok ->
+            filezcache_entry_manager:delete(Key);
+        {error, _} = Error ->
+            Error
```

Comment:
Ah yes, I think that in this case `noproc` could be handled as `ok` and we could proceed with the `delete(Key)`. It might happen on race conditions between the `lookup/1` and the process exiting.